include/exadg/acoustic_conservation_equations/spatial_discretization/operators/operator.h

```diff
@@ -0,0 +1,493 @@
+  {
+    do_evaluate(dst, src, true, time);
+  }
+
+  void
+  evaluate_add(BlockVectorType & dst, BlockVectorType const & src, double const time) const
+  {
+    do_evaluate(dst, src, false, time);
+  }
+
+private:
+  void
+  do_evaluate(BlockVectorType &       dst,
+              BlockVectorType const & src,
+              double const            time,
+              bool const              zero_dst_vector) const
+  {
+    evaluation_time = (Number)time;
+
+    matrix_free->loop(&This::cell_loop,
+                      &This::face_loop,
+                      &This::boundary_face_loop,
+                      this,
+                      dst,
+                      src,
+                      zero_dst_vector,
+                      dealii::MatrixFree<dim, Number>::DataAccessOnFaces::values,
+                      dealii::MatrixFree<dim, Number>::DataAccessOnFaces::values);
+  }
+
+  void
+  do_cell_integral(CellIntegratorP & pressure, CellIntegratorU & velocity) const
+  {
+    if(data.formulation == Formulation::Weak)
+    {
+      for(unsigned int q : pressure.quadrature_point_indices())
+      {
+        vector const flux_momentum = kernel.get_volume_flux_weak_momentum(velocity, q);
+        scalar const flux_mass     = kernel.get_volume_flux_weak_mass(pressure, q);
+
+        pressure.submit_gradient(rhocc * flux_momentum, q);
+        velocity.submit_divergence(rho_inv * flux_mass, q);
+      }
+    }
+    else if(data.formulation == Formulation::Strong)
+    {
+      for(unsigned int q : pressure.quadrature_point_indices())
+      {
+        scalar const flux_momentum = kernel.get_volume_flux_strong_momentum(velocity, q);
+        vector const flux_mass     = kernel.get_volume_flux_strong_mass(pressure, q);
+
+        pressure.submit_value(rhocc * flux_momentum, q);
+        velocity.submit_value(rho_inv * flux_mass, q);
+      }
+    }
+    else if(data.formulation == Formulation::SkewSymmetric)
+    {
+      for(unsigned int q : pressure.quadrature_point_indices())
+      {
+        vector const flux_momentum = kernel.get_volume_flux_weak_momentum(velocity, q);
+        vector const flux_mass     = kernel.get_volume_flux_strong_mass(pressure, q);
+
+        pressure.submit_gradient(rhocc * flux_momentum, q);
+        velocity.submit_value(rho_inv * flux_mass, q);
+      }
+    }
+    else
+    {
+      AssertThrow(false, dealii::ExcMessage("Not implemented."));
+    }
+  }
+
+
+  template<bool weight_neighbor, // = false for cell centric loops and boundary loops
+           typename ExteriorFaceIntegratorP,
+           typename ExteriorFaceIntegratorU>
+  void
+  do_face_integral(FaceIntegratorP &         pressure_m,
+                   ExteriorFaceIntegratorP & pressure_p,
+                   FaceIntegratorU &         velocity_m,
+                   ExteriorFaceIntegratorU & velocity_p) const
+  {
+    for(unsigned int q : pressure_m.quadrature_point_indices())
+    {
+      scalar const pm = pressure_m.get_value(q);
```

Comment:
In case of implicit problems with the solution of a linear system of equations, we need to be able to compute only the inhomogeneous action of the operator on boundary faces. This means that `pressure_m.get_value(q)` would not be sufficient, but we would need something like `calculate_interior_value(...)`, see ConvDiff module: https://github.com/exadg/exadg/blob/51a6a7e65e81c4a3816367da24f36a4d8fc813b9/include/exadg/convection_diffusion/spatial_discretization/operators/weak_boundary_conditions.h#L49-L69

@jh66637 how would you realize this? Following the design of the present PR, we could also template on the interior face integrator, increasing the number of template parameters by two. However, for boundary faces, this means we would have to define not only a class `ExteriorBoundaryFaceIntegrator`, but also a class `InteriorBoundaryFaceIntegrator`.